flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveLookupTableSource.java

```diff
@@ -114,16 +117,16 @@ private void validateLookupConfigurations() {
 			Duration monitorInterval = configuration.get(STREAMING_SOURCE_MONITOR_INTERVAL) == null
 					? DEFAULT_LOOKUP_MONITOR_INTERVAL
 					: configuration.get(STREAMING_SOURCE_MONITOR_INTERVAL);
-			Preconditions.checkArgument(
-					monitorInterval.toMillis() >= DEFAULT_LOOKUP_MONITOR_INTERVAL.toMillis(),
-					String.format(
-							"Currently the value of '%s' is required bigger or equal to default value '%s' " +
-									"when set '%s' to 'latest', but actual is '%s'",
-							STREAMING_SOURCE_MONITOR_INTERVAL.key(),
-							DEFAULT_LOOKUP_MONITOR_INTERVAL.toMillis(),
-							STREAMING_SOURCE_PARTITION_INCLUDE.key(),
-							monitorInterval.toMillis())
-			);
+
+			if (monitorInterval.toMillis() < DEFAULT_LOOKUP_MONITOR_INTERVAL.toMillis()) {
+				LOG.warn(String.format(
+					"Currently the recommended value of '%s' is bigger than default value '%s' " +
```

Comment:
I'm curious how we decided that 1h is the recommended value for monitor interval, and why a smaller value, say, 30min can cause pressure on metastore?

Reply:
It seems to me that contacting metastore every 30min for new partitions should be absolutely fine.

Reply:
The duration `1h` is an empirical value, compare to `30min`, hive partition that use hour is more common that hive partition use  half an hour.

Reply:
I agree `1h` may be more popular than `30min` for partitioning. But just because more users use `1h` doesn't mean those using `30min` deserve a scary warning.

When writing stream data into hive, we usually tell users that the minimum value supported for `sink.partition-commit.delay` is around `10min`. So I suppose the warning threshold here shouldn't be bigger than that. Because if users can commit new partitions every 10min, it's reasonable to assume they also want to monitor at roughly the same interval. WDYT?

Reply:
I agree with @lirui-apache here.

Reply:
>So I suppose the warning threshold here shouldn't be bigger than that.

I totally agree that we should align the behaviour, but as the document reminded, but currently the look up hive parttiont implementation is visit metastore from every task and the  `hive-streaming source/sink` just visit metastore from one task. So, I'm worried about use same value is not a good choice from current situation.

Reply:
@leonardBang  I see your point and I didn't realize it's distributed access to HMS. In that case, I'm fine to keep the threshold as it is. Perhaps we should worry more about how many tasks can concurrently access HMS, than how often they access.

One more suggestion about the log message: I think `"Currently the recommended value of '%s' is at least '%s' when set...` is easier to understand.

Reply:
@lirui-apache Nice suggestion, will update soon

Reply:
> Perhaps we should worry more about how many tasks can concurrently access HMS, than how often they access.

Maybe we need introduce a new mechanism the source can notify the all join subtask when to update in Flink SQL